python_modules/dagster/dagster/_core/definitions/asset_condition/asset_condition.py

```diff
@@ -439,17 +397,20 @@ def parents_updated_since_cron(cron_schedule: str, timezone: str = "UTC") -> "As
         from ..auto_materialize_rule import AutoMaterializeRule
 
         return ~RuleCondition(
-            AutoMaterializeRule.skip_on_not_all_parents_updated_since_cron(cron_schedule, timezone)
+            rule=AutoMaterializeRule.skip_on_not_all_parents_updated_since_cron(
+                cron_schedule, timezone
+            )
         )
 
 
 @experimental
 @whitelist_for_serdes
-@dataclass(frozen=True)
 class RuleCondition(AssetCondition):
     """This class represents the condition that a particular AutoMaterializeRule is satisfied."""
 
-    rule: "AutoMaterializeRule"
+    from ..auto_materialize_rule import AutoMaterializeRule
```

Comment:
pydantic doesn't like forward refs

Reply:
huh, docs make it seem like this should be fine